northstar_tests/src/runtime.rs

```diff
@@ -75,36 +79,42 @@ impl Runtime {
 
     /// Launches an instance of north with the specified configuration
     pub async fn launch_with_config(mut config: Config) -> Result<Runtime> {
-        // It is expected that the "default" repository contains the example containers
-        let default_repository = &mut config
-            .repositories
-            .get_mut("default")
-            .ok_or_else(|| eyre!("No \"default\" repository in configuration"))?
-            .dir;
-
-        // Discard changes to the default repository at the end of the test
-        *default_repository =
-            tokio::task::block_in_place(|| overlay_directory(default_repository))?;
+        // Test scoped repository
+        let repository = TempDir::new()?.into_path();
```

Comment:
This persists the TempDir on disc and it will never be deleted:
https://docs.rs/tempfile/3.2.0/tempfile/struct.TempDir.html#method.into_path

Reply:
it is deleted in the shutdown

Reply:
What happens with a panic?

A race in the Drop of Runtime sounds suspicious. The `Drop` for `TempDir` does: https://docs.rs/tempfile/3.2.0/src/tempfile/dir.rs.html#393-400.

The TempDir should be also free from opened fds.

Reply:
It should yes, but it fails unfortunately. My guess would be that the fs is not immediately synced after the drop but I'm not sure.

Reply:
Did you check which fd make the leak test fail? This would be interesting.

Reply:
It is definitely related to the test directory, can't 100% be sure because it's just a number. I will try to have an other look at this.

Reply:
> It is definitely related to the test directory, can't 100% be sure because it's just a number. I will try to have an other look at this.

Add a long sleep before the `Drop` of the Tmpdir and do a `ls -la` in /proc/<PID>/fd` with and without the tempdir. This should point you to right thing.